src/main/kotlin/com/terraformation/backend/customer/db/DisclaimerStore.kt

```diff
@@ -0,0 +1,123 @@
+import jakarta.inject.Named
+import java.time.Instant
+import java.time.InstantSource
+import org.jooq.DSLContext
+import org.jooq.impl.DSL
+
+@Named
+class DisclaimerStore(
+    private val clock: InstantSource,
+    private val disclaimersDao: DisclaimersDao,
+    private val dslContext: DSLContext,
+    private val userDisclaimersDao: UserDisclaimersDao,
+) {
+
+  fun createDisclaimer(content: String, effectiveOn: Instant): DisclaimerId {
+    requirePermissions { manageDisclaimers() }
+
+    val row =
+        DisclaimersRow(
+            content = content,
+            effectiveOn = effectiveOn,
+        )
+
+    disclaimersDao.insert(row)
+    return row.id!!
+  }
+
+  fun deleteDisclaimer(disclaimerId: DisclaimerId) {
+    requirePermissions { manageDisclaimers() }
+
+    disclaimersDao.deleteById(disclaimerId)
+  }
+
+  fun fetchAllDisclaimers(): List<DisclaimerModel> {
+    requirePermissions { manageDisclaimers() }
+
+    val usersMultiset =
+        with(USER_DISCLAIMERS) {
+          DSL.multiset(
+                  DSL.select(
+                          USER_ID,
+                          ACCEPTED_ON,
+                      )
+                      .from(this)
+                      .where(DISCLAIMER_ID.eq(DISCLAIMERS.ID)))
+              .convertFrom { result ->
+                result.associate { record -> record[USER_ID]!! to record[ACCEPTED_ON]!! }
+              }
+        }
+
+    return with(DISCLAIMERS) {
+      dslContext
+          .select(
+              ID,
+              CONTENT,
+              EFFECTIVE_ON,
+              usersMultiset,
+          )
+          .from(this)
+          .fetch { DisclaimerModel.of(it, usersMultiset) }
+    }
+  }
+
+  /**
+   * Fetch the latest effective disclaimer. Omits future disclaimers. Null if no effective
+   * disclaimer exists.
+   */
+  fun fetchCurrentDisclaimer(): DisclaimerModel? {
+    requirePermissions { readCurrentDisclaimer() }
+
+    return with(DISCLAIMERS) {
+      dslContext
+          .select(
+              ID,
+              CONTENT,
+              EFFECTIVE_ON,
+              USER_DISCLAIMERS.ACCEPTED_ON,
+          )
+          .from(this)
+          .leftJoin(USER_DISCLAIMERS)
+          .on(USER_DISCLAIMERS.DISCLAIMER_ID.eq(ID))
+          .and(USER_DISCLAIMERS.USER_ID.eq(currentUser().userId))
+          .where(DISCLAIMERS.EFFECTIVE_ON.le(clock.instant()))
+          .orderBy(EFFECTIVE_ON.desc())
+          .limit(1)
+          .fetchOne { DisclaimerModel.of(it) }
+    }
+  }
+
+  /** Accepts the disclaimer */
+  fun acceptCurrentDisclaimer() {
+    requirePermissions { acceptCurrentDisclaimer() }
+
+    val currentDisclaimer = fetchCurrentDisclaimer()
+
+    if (currentDisclaimer == null) {
+      throw IllegalStateException("No current disclaimer exists.")
+    }
+
+    if (currentDisclaimer.acceptedOn != null) {
+      throw IllegalStateException("User has already accepted the current disclaimer.")
+    }
```

Comment:
Maybe we should treat this as a no-op instead. Log an info message, perhaps, but don't treat the request as failed.